app/celery/v3/notification_tasks.py

```diff
@@ -105,21 +137,24 @@ def v3_process_notification(request_data: dict, service_id: str, api_key_id: str
         v3_send_email_notification.delay(notification, template)
     elif notification.notification_type == SMS_TYPE:
         if notification.sms_sender_id is None:
-            # Get the template or service default sms_sender_id.
-            # TODO
-            v3_persist_failed_notification(
-                notification, NOTIFICATION_TECHNICAL_FAILURE, "Default logic for sms_sender_id is not yet implemented."
-            )
-            return
+            err, sms_sender_id = get_default_sms_sender_id(service_id)
+            if err is not None:
+                v3_persist_failed_notification(notification, NOTIFICATION_TECHNICAL_FAILURE, err)
+                return
+            notification.sms_sender_id = sms_sender_id
 
         # TODO - Catch db connection errors and retry?
-        query = select(ServiceSmsSender).where(ServiceSmsSender.id == request_data["sms_sender_id"])
+        query = select(ServiceSmsSender).where(
+            (ServiceSmsSender.id == notification.sms_sender_id) &
+            (ServiceSmsSender.service_id == service_id)
+        )
```

Comment:
If get_default_sms_sender_id got called, this query is getting the same ServiceSmsSender instance.  It would be better to have that function return the instance instead of just the ID so we don't always need another db query here.